lightning/src/util/logger.rs

```diff
@@ -139,6 +147,26 @@ impl<'a> Record<'a> {
 pub trait Logger {
 	/// Logs the `Record`
 	fn log(&self, record: &Record);
+	/// This method will be called on the logger and expects it to return
+	/// the peer_id pertaining to the log.
+	#[doc(hidden)]
+	fn peer_id(&self) -> Option<PublicKey> { None }
```

Comment:
I played around with this. Unfortunately it doesnt seem v possible.

tried `trait ContextLogger: Logger` 
- this doesnt "extend" Logger, so it did not work
- doesnt work for trait objects (dyn Logger) which we use in the style of `Arc::<Logger>`

tried `impl<T:Logger> LoggerContextExt for T` 
- also doesnt work for trait objects. 
![image](https://github.com/lightningdevkit/rust-lightning/assets/906931/d594b3ac-f4c9-4357-9017-c9b6845601a8)
- error was `the method `peer_id` exists for reference `&dyn Logger`, but its trait bounds were not satisfied
method cannot be called on `&dyn Logger` due to unsatisfied trait bounds` 
The reason for this error is that trait objects do not have access to trait extensions or additional methods defined outside of the original trait. When you create a trait object (&dyn Trait or Box<dyn Trait>), the type information for the concrete type implementing the trait is erased, and you can only access the methods defined in the original trait itself. 
it seems that the vtable of the trait object isnt populated with the trait extension methods and that isnt supported right now. 
- Moreover, blanket implementations meant that individual struct implementations for LoggerContextExt were not possible

One way we could possibly do this is to re-wrap any incoming Loggers with a default WithContext. 
let logger = WithContext { logger: &self.logger, peer_id: None, channel_id: None};
technically possible in theory, but seems impractical. 

let me know if i missed something @TheBlueMatt @jkczyz

Reply:
> just remove the new Logger methods and instead have the WithContext loggers simply wrap underlying loggers and modify the Record as it passes through them.

im not completely sure how this is supposed to look like. currently the records are only created upon the log_xx macro and within the macro logger, so we're not passing Record objects around. 

> One way we could possibly do this is to re-wrap any incoming Loggers with a default WithContext.
let logger = WithContext { logger: &self.logger, peer_id: None, channel_id: None};
technically possible in theory, but seems impractical.

if you are saying that we should wrap _all_ the loggers, im trying this out and this will probably make _every_ log line need a change or changes to some of the internal method signatures. not really sure if this is the right direction we want to take. 

maybe you can provide me with an example of what you mean? @TheBlueMatt

Reply:
okay thanks! sounds good, let me make those changes. and update the docs too!